src/Aspire.Hosting/Dashboard/AspireDashboardResourceBuilderExtensions.cs

```diff
@@ -0,0 +1,60 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for creating Aspire Dashboard resources in the application model.
+/// </summary>
+public static class AspireDashboardResourceBuilderExtensions
+{
+    /// <summary>
+    /// Creates a new Aspire Dashboard resource builder with the specified name.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/> instance.</param>
+    /// <param name="name">The name of the Aspire Dashboard resource.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{AspireDashboardResource}"/>.</returns>
+    /// <remarks>
+    /// This method initializes a new Aspire Dashboard resource that can be used to visualize telemetry data
+    /// in the Aspire Hosting environment. The resource is not automatically added to the application model;
+    /// instead, it returns a builder that can be further configured and added as needed.
+    /// </remarks>
+    /// <exception cref="ArgumentNullException">Thrown when <paramref name="builder"/> is <c>null</c>.</exception>
+    /// <exception cref="ArgumentException">Thrown when <paramref name="name"/> is <c>null</c> or empty.</exception>
+    public static IResourceBuilder<AspireDashboardResource> CreateDashboard(
```

Comment:
I'm wondering whether we should expose this method. Is dashboard hosting unique enough to each compute environment (especially once we look at resource servers). For example, with ACA deployments you could have WithDashboard(false) but the implementation would be quite different to what you would do for Docker Compose or Kubernetes.

Reply:
Moved to the docker assembly and renamed.